cpu/stm32f1/periph/gpio.c

```diff
@@ -30,1015 +30,190 @@
-        case GPIO_11:
-            GPIO_11_PORT->BSRR = (1 << GPIO_11_PIN);
-            break;
-#endif
-#if GPIO_12_EN
-        case GPIO_12:
-            GPIO_12_PORT->BSRR = (1 << GPIO_12_PIN);
-            break;
-#endif
-#if GPIO_13_EN
-        case GPIO_13:
-            GPIO_13_PORT->BSRR = (1 << GPIO_13_PIN);
-            break;
-#endif
-#if GPIO_14_EN
-        case GPIO_14:
-            GPIO_14_PORT->BSRR = (1 << GPIO_14_PIN);
-            break;
-#endif
-#if GPIO_15_EN
-        case GPIO_15:
-            GPIO_15_PORT->BSRR = (1 << GPIO_15_PIN);
-            break;
-#endif
-    }
+    _port(pin)->BSRR = (1 << _pin_num(pin));
 }
 
-void gpio_clear(gpio_t dev)
+void gpio_clear(gpio_t pin)
 {
-    switch (dev) {
-#if GPIO_0_EN
-        case GPIO_0:
-            GPIO_0_PORT->BRR = (1 << GPIO_0_PIN);
-            break;
-#endif
-#if GPIO_1_EN
-        case GPIO_1:
-            GPIO_1_PORT->BRR = (1 << GPIO_1_PIN);
-            break;
-#endif
-#if GPIO_2_EN
-        case GPIO_2:
-            GPIO_2_PORT->BRR = (1 << GPIO_2_PIN);
-            break;
-#endif
-#if GPIO_3_EN
-        case GPIO_3:
-            GPIO_3_PORT->BRR = (1 << GPIO_3_PIN);
-            break;
-#endif
-#if GPIO_4_EN
-        case GPIO_4:
-            GPIO_4_PORT->BRR = (1 << GPIO_4_PIN);
-            break;
-#endif
-#if GPIO_5_EN
-        case GPIO_5:
-            GPIO_5_PORT->BRR = (1 << GPIO_5_PIN);
-            break;
-#endif
-#if GPIO_6_EN
-        case GPIO_6:
-            GPIO_6_PORT->BRR = (1 << GPIO_6_PIN);
-            break;
-#endif
-#if GPIO_7_EN
-        case GPIO_7:
-            GPIO_7_PORT->BRR = (1 << GPIO_7_PIN);
-            break;
-#endif
-#if GPIO_8_EN
-        case GPIO_8:
-            GPIO_8_PORT->BRR = (1 << GPIO_8_PIN);
-            break;
-#endif
-#if GPIO_9_EN
-        case GPIO_9:
-            GPIO_9_PORT->BRR = (1 << GPIO_9_PIN);
-            break;
-#endif
-#if GPIO_10_EN
-        case GPIO_10:
-            GPIO_10_PORT->BRR = (1 << GPIO_10_PIN);
-            break;
-#endif
-#if GPIO_11_EN
-        case GPIO_11:
-            GPIO_11_PORT->BRR = (1 << GPIO_11_PIN);
-            break;
-#endif
-#if GPIO_12_EN
-        case GPIO_12:
-            GPIO_12_PORT->BRR = (1 << GPIO_12_PIN);
-            break;
-#endif
-#if GPIO_13_EN
-        case GPIO_13:
-            GPIO_13_PORT->BRR = (1 << GPIO_13_PIN);
-            break;
-#endif
-#if GPIO_14_EN
-        case GPIO_14:
-            GPIO_14_PORT->BRR = (1 << GPIO_14_PIN);
-            break;
-#endif
-#if GPIO_15_EN
-        case GPIO_15:
-            GPIO_15_PORT->BRR = (1 << GPIO_15_PIN);
-            break;
-#endif
-    }
+    _port(pin)->BRR = (1 << _pin_num(pin));
 }
 
-
-void gpio_toggle(gpio_t dev)
+void gpio_toggle(gpio_t pin)
 {
-    if (gpio_read(dev)) {
-        gpio_clear(dev);
+    if (gpio_read(pin)) {
+        gpio_clear(pin);
     }
     else {
-        gpio_set(dev);
+        gpio_set(pin);
     }
 }
 
-void gpio_write(gpio_t dev, int value)
+void gpio_write(gpio_t pin, int value)
```

Comment:
I never really understood why we need this function.. Anyway, it's not related to this PR